### PR TITLE
PHP 7 strict typing, PSR compliance, code cleanup

### PR DESCRIPTION
**Pull request by jissereitsma**
Some fixes to add strict typing to all PHP files, plus some argument hinting and return type hinting to get code compliant with PHP 7 standards. Also some fixes for PSR compliance and code cleanup (removing elses, imports of namespaces).

## This PR is a:

[ ] New feature
[X ] Enhancement/Optimization
[ ] Refactor
[ ] Bugfix
[ ] Test for existing code
[ ] Documentation

### CODE REVIEW

File: packages/pwa-module/etc/module.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0"?>
-
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
     <module name="Magento_Pwa" setup_version="1.0.0">
+        <sequence>
+            <module name="Magento_Backend" />
+            <module name="Magento_Store" />
+        </sequence>
     </module>
 </config>
```
Comment (davemacaulay) [inline packages/pwa-module/etc/module.xml]: Whilst we're here, let's add a new line to this file.

File: packages/pwa-module/registration.php [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
 
-\Magento\Framework\Component\ComponentRegistrar::register(
-    \Magento\Framework\Component\ComponentRegistrar::MODULE,
+use Magento\Framework\Component\ComponentRegistrar;
+
+ComponentRegistrar::register(
+    ComponentRegistrar::MODULE,
     'Magento_Pwa',
     __DIR__
 );
```
Comment (davemacaulay) [inline packages/pwa-module/registration.php]: _I hate new line comments_

File: packages/pwa-module/view/frontend/layout/default.xml [hunk 3]
```diff
@@ -1,13 +1,8 @@
...
     <body>
         <referenceContainer name='root'>
-            <block
-                class='Magento\Pwa\Block\Bundle'
-                name='Magento_Pwa_Block_Bundle'
-            />
+            <block class='Magento\Pwa\Block\Bundle' name='Magento_Pwa_Block_Bundle' />
         </referenceContainer>
     </body>
 </page>
```
Comment (davemacaulay) [inline packages/pwa-module/view/frontend/layout/default.xml]: 🤕 _sorry!_

File: packages/pwa-module/view/frontend/layout/default.xml [hunk 4]
```diff
@@ -1,13 +1,8 @@
...
-    xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'
-    xsi:noNamespaceSchemaLocation='urn:magento:framework:View/Layout/etc/page_configuration.xsd'>
+<page xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xsi:noNamespaceSchemaLocation='urn:magento:framework:View/Layout/etc/page_configuration.xsd'>
     <body>
         <referenceContainer name='root'>
-            <block
-                class='Magento\Pwa\Block\Bundle'
-                name='Magento_Pwa_Block_Bundle'
-            />
+            <block class='Magento\Pwa\Block\Bundle' name='Magento_Pwa_Block_Bundle' />
```
Comment (davemacaulay) [inline packages/pwa-module/view/frontend/layout/default.xml]: We should use double quotes throughout XML files.

File: packages/pwa-module/Helper/WebpackConfig.php [hunk 5]
```diff
@@ -59,7 +76,7 @@ class WebpackConfig implements \JsonSerializable
     const DEFAULT_SERVICE_WORKER_NAME = 'sw.js';
 
     /**
-     * @var \Magento\Framework\View\Config
+     * @var Config
      */
     private $_viewConfig;
```
Comment (davemacaulay) [inline packages/pwa-module/Helper/WebpackConfig.php]: We shouldn't use `_` to indicate protected or private visibility. _Appreciate this was existing, but nice to fix in here_

https://www.php-fig.org/psr/psr-2/#42-properties

**Comment by zetlen** [comment]
Thanks so much for your attention to this, @jissereitsma and @davemacaulay. Full disclosure: the current `pwa-module` code is probably not going to survive, beyond a couple of core concepts. But we will still track and ultimately merge this, as a key example of PHP compliance auditing for this repo.

File: packages/pwa-module/etc/module.xml (hunk 1 above)
Comment (jissereitsma) [reply packages/pwa-module/etc/module.xml]: Hmm, within PHP, newlines at the end of the file are part of the PSR standard. However, does this also apply to XML. It is simply the first time I heard so ...

File: packages/pwa-module/view/frontend/layout/default.xml (hunk 3 above)
Comment (jissereitsma) [reply packages/pwa-module/view/frontend/layout/default.xml]: You're sorry about the PR or sorry about having a different opinion? :)

File: packages/pwa-module/view/frontend/layout/default.xml (hunk 4 above)
Comment (jissereitsma) [reply packages/pwa-module/view/frontend/layout/default.xml]: Completely agree. Will improve PR soon.

File: packages/pwa-module/Helper/WebpackConfig.php (hunk 5 above)
Comment (jissereitsma) [reply packages/pwa-module/Helper/WebpackConfig.php]: Completely agree. Will improve PR soon.

**Comment by jissereitsma** [comment]
@zetlen I completely get that. And I actually guessed that this whole module was temporary anyway. It's just to make sure that other PHP devs that come along, do not start bashing the whole PWA Studio project just because the PHP code was not 100% compliant.

File: packages/pwa-module/registration.php (hunk 2 above)
Comment (jissereitsma) [reply packages/pwa-module/registration.php]: https://github.com/magento-research/pwa-studio/pull/105/commits/2fc1eb9f9c300cb3e944b71c9696b0d1e17a9a4e

File: packages/pwa-module/view/frontend/layout/default.xml (hunk 4 above)
Comment (jissereitsma) [reply packages/pwa-module/view/frontend/layout/default.xml]: https://github.com/magento-research/pwa-studio/pull/105/commits/a996e4f6b1386151ea51deb446834302c42d63d3

File: packages/pwa-module/Helper/WebpackConfig.php (hunk 5 above)
Comment (jissereitsma) [reply packages/pwa-module/Helper/WebpackConfig.php]: https://github.com/magento-research/pwa-studio/pull/105/commits/bbece38eec039ca0bab1591cc70e372f4d241e63

**Comment by jissereitsma** [comment]
@davemacaulay I've updated things now with your suggestions. However, I see a lot more that could be improved: Helpers are silly, should be refactored into SOLID classes. The `Result` in the `Model` folder is better placed in a `Result` folder instead. Etc, etc. But for now, I think this little cleanup was nice enough. As @zetlen said, it is just a temporary thing.

File: packages/pwa-module/etc/module.xml (hunk 1 above)
Comment (davemacaulay) [reply packages/pwa-module/etc/module.xml]: Internally we typically just add new lines to everything for consistency. Unsure if with XML it's part of any convention though.

**Comment by davemacaulay** [comment]
@jissereitsma I agree! My comments just bought us a bit closer to being inline with our internal static tests. Thanks for the changes.

**Comment by DrewML** [comment]
Thanks @jissereitsma!

**Pull request merged**